File: week-1/02-dive-into-python/02-integer-prime-factorization.js






function prime_factorization(n) {
	let prime_numbers = [2, 3, 5, 7];
	let output = [];
	for(let divider of prime_numbers){
		let result = [];
		if (n % divider === 0) {
			result = recursion(n, divider, 0);
			n = result[0];
			let counter = result[1];
			let arr = [divider, counter];
			output.push(arr);
		}
	}

	if(n > 1){
		let counter = 1;
		let arr = [n, counter];
		output.push(arr);
	}
	

	return output;
}

console.log(prime_factorization(14));





function recursion(n, divider, counter){
	if (n % divider === 0) {
		n = n / divider;
		counter++;
		[n, counter] = recursion(n, divider, counter);
	}

	return [n, counter];
}

File: week-1/02-dive-into-python/02-integer-prime-factorization.py
# Integer prime factorization
def recursion(n, divider, counter):
    if n % divider == 0:
        n /= divider
        counter += 1
        outp = recursion(n, divider, counter)
        n = outp[0]
        counter = outp[1]

    return [n, counter]




def prime_factorization(n):
    prime_numbers = [2, 3, 5, 7]
    output = []
    for divider in prime_numbers:
        if n % divider == 0:
            outp = recursion(n, divider, 0)
            n = outp[0]
            counter = outp[1]
            ls = (divider, counter)
            output.append(ls)
    if n > 1:
        output.append((n, 1))
    print(output)


prime_factorization(1000)
